debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/connection/pgoutput/PgOutputStreamingMessageDecoderIT.java

```diff
@@ -0,0 +1,811 @@
+    private static final long MEDIUM_TRANSACTION_SIZE = 100;
+    private static final long LARGE_TRANSACTION_SIZE = 1000;
+    private static final long EXTRA_LARGE_TRANSACTION_SIZE = 10000;
+    private static final long STREAMING_THRESHOLD_SIZE = 2000; // Size that definitely triggers streaming
+    private static final long MASSIVE_STREAMING_SIZE = 5000; // Very large streaming transaction
+
+    @Before
+    public void before() throws Exception {
+        setupSystemProperties();
+        setupDatabase();
+        initializeConnectorTestFramework();
+    }
+
+    @After
+    public void after() throws Exception {
+        resetSystemConfiguration();
+        cleanupDatabaseState();
+    }
+
+    @Test
+    public void testSnapshotIsWorking_whenSmallTransactionsHaveExecuted_shouldAbleToProcessAllData() throws InterruptedException {
+        String threadName = Thread.currentThread().getName();
+        executeInsertSQLStatements(threadName, SMALL_TRANSACTION_SIZE);
+
+        startConnectorWithInitialSnapshot();
+
+        validateRecordsForSingleThread(threadName, SMALL_TRANSACTION_SIZE);
+    }
+
+    @Test
+    public void testSnapshotIsWorking_whenLargeTransactionsHaveExecuted_shouldAbleToProcessAllData() throws InterruptedException {
+        String threadName = Thread.currentThread().getName();
+        executeInsertSQLStatements(threadName, LARGE_TRANSACTION_SIZE);
+
+        startConnectorWithInitialSnapshot();
+
+        validateRecordsForSingleThread(threadName, LARGE_TRANSACTION_SIZE);
+    }
+
+    @Test
+    public void testStreamingIsWorking_whenSmallTransactionsAreExecuting_shouldAbleToProcessAllData() throws InterruptedException {
+        startConnectorWithInitialSnapshot();
+
+        String threadName = Thread.currentThread().getName();
+        executeInsertSQLStatements(threadName, SMALL_TRANSACTION_SIZE);
+
+        validateRecordsForSingleThread(threadName, SMALL_TRANSACTION_SIZE);
+    }
+
+    @Test
+    public void testStreamingIsWorking_whenLargeInProgressTransactionsAreExecuting_shouldAbleToProcessAllStreamedData() throws InterruptedException {
+        startConnectorWithInitialSnapshot();
+
+        String threadName = Thread.currentThread().getName();
+        executeInsertSQLStatements(threadName, LARGE_TRANSACTION_SIZE);
+
+        validateRecordsForSingleThread(threadName, LARGE_TRANSACTION_SIZE);
+    }
+
+    @Test
+    public void testStreamingIsWorking_whenSmallTransactionsAreExecutingAfterSnapshot_shouldAbleToProcessAllData() throws InterruptedException {
+        String threadName = Thread.currentThread().getName();
+
+        // Execute transactions before connector start
+        executeInsertSQLStatements(threadName, SMALL_TRANSACTION_SIZE);
+
+        startConnectorWithInitialSnapshot();
+
+        // Execute transactions after connector start
+        executeInsertSQLStatements(threadName, SMALL_TRANSACTION_SIZE);
+
+        Thread.sleep(5000);
```

Comment:
Sleeps are brittle in tests. Is not it possible to rewrite it using `Awaitility` condition?
Same for the rest of tests.